### PR TITLE
HCE-795: Add Project ID to Packer

### PR DESCRIPTION
**Pull request by paraspraj**
### :hammer_and_wrench: Description

Add project_id to Packer datasources and resources

### CODE REVIEW

File: internal/provider/resource_packer_channel.go [hunk 1]
```diff
@@ -95,11 +104,6 @@ func resourcePackerChannel() *schema.Resource {
 				Type:        schema.TypeString,
 				Computed:    true,
 			},
-			"project_id": {
-				Description: "The ID of the project this HCP Packer registry is located in.",
```
Comment (nywilken) [inline internal/provider/resource_packer_channel.go:99]: The field descriptions within this resource appear to be behind a commit https://github.com/hashicorp/terraform-provider-hcp/blob/0d79f9c8040747a24a24df75330f3ad47c9f2f57/internal/provider/resource_packer_channel.go#L101

Is your local branch up to date with what is on main?

**Review by nywilken — CHANGES_REQUESTED** [review]
This change always looks good, but it appears that the resource code may be out of date with whats on main. Is this change to support multi-projects per org?

File: internal/provider/resource_packer_channel.go (hunk 1 above)
Comment (bcmdarroch) [reply internal/provider/resource_packer_channel.go:99]: This PR is going into a feature branch, which I've just rebased: https://github.com/hashicorp/terraform-provider-hcp/pull/454. Hopefully that addresses the issue!

Comment (bcmdarroch) [reply internal/provider/resource_packer_channel.go]: Ah, and just pushed the rebase of this branch. Do the latest changes capture all the fixes you were expecting? I referred to this change:
https://github.com/hashicorp/terraform-provider-hcp/commit/c1730879d133ea105c34314806cab01000271a69

Comment (bcmdarroch) [reply internal/provider/resource_packer_channel.go]: Haha I might have caused more issues with my rebase 😅 Standby.

Comment (paraspraj) [reply internal/provider/resource_packer_channel.go]: Hey @nywilken! Wanted to check in to see if you were okay with the changes that were made to bring this PR back up to `main`.

**Review by nywilken — APPROVED** [review]
Changes look good to me. Thanks for rebasing.

```ok      github.com/hashicorp/terraform-provider-hcp/internal/input      0.265s [no tests to run]
=== RUN   TestAcc_dataSourcePacker
--- PASS: TestAcc_dataSourcePacker (11.19s)
=== RUN   TestAcc_dataSourcePacker_revokedIteration
--- PASS: TestAcc_dataSourcePacker_revokedIteration (13.75s)
=== RUN   TestAcc_dataSourcePackerImage
--- PASS: TestAcc_dataSourcePackerImage (9.87s)
=== RUN   TestAcc_dataSourcePackerImage_revokedIteration
--- PASS: TestAcc_dataSourcePackerImage_revokedIteration (12.35s)
=== RUN   TestAcc_dataSourcePackerImage_channelAndIterationIDReject
--- PASS: TestAcc_dataSourcePackerImage_channelAndIterationIDReject (4.25s)
=== RUN   TestAcc_dataSourcePackerImage_channelAccept
--- PASS: TestAcc_dataSourcePackerImage_channelAccept (7.58s)
=== RUN   TestAcc_dataSourcePackerIteration
--- PASS: TestAcc_dataSourcePackerIteration (7.53s)
=== RUN   TestAcc_dataSourcePackerIteration_revokedIteration
--- PASS: TestAcc_dataSourcePackerIteration_revokedIteration (12.66s)
=== RUN   TestAccPackerChannel
--- PASS: TestAccPackerChannel (8.25s)
=== RUN   TestAccPackerChannel_AssignedIteration
--- PASS: TestAccPackerChannel_AssignedIteration (10.31s)
=== RUN   TestAccPackerChannel_UpdateAssignedIteration
--- PASS: TestAccPackerChannel_UpdateAssignedIteration (15.59s)
=== RUN   TestAccPackerChannel_UpdateAssignedIterationWithFingerprint
--- PASS: TestAccPackerChannel_UpdateAssignedIterationWithFingerprint (7.92s)
PASS
ok      github.com/hashicorp/terraform-provider-hcp/internal/provider   121.591s
```

**Pull request merged**